adr/0024-starlight.md

```diff
@@ -0,0 +1,76 @@
+# 24. Migrate documentation to Starlight
+
+Date: 2024-03-30
+
+## Status
+
+Accepted
+
+## Context
+
+The current documentation site is built with Docusaurus 2. While Docusaurus has served us well, it has some limitations, primarily around performance and dependency management.
+
+### Proposed Solutions
+
+1. [**Starlight**](https://starlight.astro.build/) is a new documentation site generator that is built with performance in mind. It is a lightweight, fast, and flexible static site generator powered by [Astro](https://astro.build) that is designed to be easy to use and extend.
+
+    - **Pros**:
+        - Fast performance.
+        - Easy to extend with custom components in any framework.
+        - Built-in support for MDX.
+        - Simplified build process.
+        - Better control over dependencies.
+        - Client-side static site search built-in.
+        - Excellent default theme.
+    - **Cons**:
+        - Newer project with a smaller ecosystem compared to others.
+        - Much fewer plugins and themes available compared to other static site generators.
+        - Only absolute URLs are supported for cross-site links (relative image URLs are supported).
+
+2. [**Hugo**](https://gohugo.io/): is another popular static site generator that is known for its speed and flexibility. It is written in Go and has a large ecosystem of themes and plugins.
+
+    - **Pros**:
+        - Fast performance.
+        - Large ecosystem of themes and plugins.
+        - By default no JavaScript is required.
+    - **Cons**:
+        - Fewer built-in features compared to others, much more of a DIY approach.
+        - Steeper learning curve compared to others.
+        - More complex build process.
+        - Not as easy to extend with custom components.
+        - Theme management is abysmal w/ a combination of Go modules, Git submodules, and NPM packages.
+        - Zero documentation themes the team liked, leading to the quandry of whether to build, and having to maintain, a custom theme.
```

Comment:
A standard documentation theme was created based on Zarf's current site and in collaboration with the Zarf team last year. We probably want to avoid every team developing a custom documentation theme as we grow our product suite.

Reply:
In this case, we are inheriting the default theme from Starlight and only tweaking theme colors. The cost of maintaining a well constructed documentation theme is normally too high for it not to have a dedicated team devoted to maintaining it.

Reply:
The theme, a fork of Google's Docsy, exists and is being used by Lula, Leapfrog, UDS, Pepr, etc. The team to support it also exists.